### ISSUE TITLE
VSTest@2 fails on login docker container with sql image

### ISSUE THREAD

**Issue by RicardoLans** [microsoft/azure-pipelines-tasks#10152]
**Question, Bug, or Feature?**  
*Type*: Question and/or Bug

**Enter Task Name**: VSTest@2

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks/VsTestV2

## Environment
- Agent - Private:
Self hosted agent based on microsoft base image.

## Issue Description
1. We build projects using
  - `NuGetRestoreV1`
  - `VSBuildV1`

2. We setup a docker container with sql server and a database from a repository in our private Azure Container Registry using:
  - `Docker@1` inputs
    - command: login
    - containerregistrytype: Azure Container Registry
    - azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
    - azureContainerRegistry: $(azureContainerRegistry)
  - `Docker@1` inputs
    - command: run
    - azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
    - azureContainerRegistry: $(azureContainerRegistry)
    - imageName: $(dockerImageName)
    - containerName: $(dockerContainerName)
    - ports: 1433

3. We use `PowerShell@2` to edit projects config files to update connectionstring to use localhost,<port_number_docker_container> as data source

4. We use `VsTestPlatformToolInstallerV1` to install latestStable from nugetOrg

[The following step fails]
5. We use `VSTest@2` to run unit tests that actually read/write data from/to the docker container sql server database:
  - `VSTest@2` inputs
    - testSelector: testAssemblies
    - testAssemblyVer2: '**\*DatabaseIntegratieTest.dll'
    - searchFolder: '$(build.artifactStagingDirectory)'
    - vsTestVersion: toolsInstaller
    - runSettingsFile: '$(build.artifactStagingDirectory)/TestSettings.runsettings'
    - codeCoverageEnabled: true
    - collectDumpOn: always

VSTest@2 is executing command:
`vstest.console.exe "E:\agent3\_work\24\a\xxx.DbTest.dll" /Settings:"E:\agent3\_work\_temp\5e7e6c40-61b1-11e9-9427-7bdc9b3786f4.runsettings" /logger:"trx" /TestAdapterPath:"E:\agent3\_work\24\a"`

Tests failing due to `System.Data.SqlClient.SqlException: Login failed for user 'sa'`

First you think 'connectionString is not correct', but we tested this exact command from commandline on our build agent itself and then everything works great. Meaning connectionString is changed correctly.

Base docker image: [microsoft/mssql-server-windows-developer](https://github.com/Microsoft/mssql-docker/tree/master/windows/mssql-server-windows-developer)
We use the base image to create a new image with our DB and schema.
For easiness we dont use a complex password for sa.

Somehow it seems that this `VSTest@2` task is not able to login (based on the error from the tests)
Also some strange (but as developer i know it sometimes can be helpfull) build information:
last succesfull build ended `2019-04-15 14:19` and first failed build started `2019-04-15 14:30` which is just 11 minutes between...
Both builds runs same task version: `2.149.6`

Please someone help me out here.
We got like 6 Pull Requests (and increasing daily) hanging on this failed build.

**Comment by nigurr** [comment]
@RicardoLans VsTest task is just wrapper around vstest.console.exe and which executes test code.
Did you try running them in local (invoking vstest.console.exe on docker image)

**Comment by RicardoLans** [comment]
@nigurr yes the exact command used by vstest task is:
`vstest.console.exe "E:\agent3\_work\24\a\xxx.DbTest.dll" /Settings:"E:\agent3\_work\_temp\5e7e6c40-61b1-11e9-9427-7bdc9b3786f4.runsettings" /logger:"trx" /TestAdapterPath:"E:\agent3\_work\24\a"`

and we ran this local on the agent and it worked
only when this command runs from our build by the VsTest task it fails

**Comment by prawalagarwal** [comment]
@RicardoLans Can you provide us logs of your failing runs after settings "system.debug" as true in the build variables?

**Comment by RicardoLans** [comment]
@prawalagarwal thank you for responding to this, i completely forgot I still had this issue on github, I disabled that specific build since we needed to complete the PR's.

I just ran the build to fullfill your request on the logs.
I noticed an updated version of the test task step and now the build is working again...
So strange but we can use it now again.

Task version before: `2.149.6` 
Task version now: `2.151.5`

Hereby closing this issue.
